src/west/commands/project.py

```diff
@@ -162,234 +162,6 @@ def do_run(self, args, user_args):
             log.inf(result)
 
 
-class Clone(WestCommand):
-    def __init__(self):
-        super().__init__(
-            'clone',
-            'clone remote projects into local installation',
-            _wrap('''
-            Clone projects.
-
-            Clones each of the specified projects (default: all projects) and
-            creates a branch in each. The branch is named after the project's
-            revision, and tracks the 'manifest-rev' branch (see below).
-
-            If the project's revision is an SHA, the branch will simply be
-            called 'work'.
-
-            This command is really just a shorthand for 'west fetch' +
-            'west checkout -b <branch name>'. If you clone a project with
-            'west fetch' instead, you will start in a detached HEAD state at
-            'manifest-rev'.
-
-            {}
-
-            {}'''.format(_NO_UPDATE_HELP, _MANIFEST_REV_HELP)))
-
-    def do_add_parser(self, parser_adder):
-        return _add_parser(
-            parser_adder, self,
-            _arg('-b',
-                 dest='branch',
-                 metavar='BRANCH_NAME',
-                 help='an alternative branch name to use, instead of one '
-                      'based on the revision'),
-            _no_update_arg,
-            _project_list_arg)
-
-    def do_run(self, args, user_args):
-        if args.update:
-            _update_west()
-
-        for project in _projects(args, listed_must_be_cloned=False):
-            if args.branch:
-                branch = args.branch
-            elif _is_sha(project.revision):
-                # Don't name the branch after an SHA
-                branch = 'work'
-            else:
-                # Use the last component of the revision, in case it is a
-                # qualified ref (refs/heads/foo and the like)
-                branch = project.revision.split('/')[-1]
-
-            _fetch(project)
-            _create_branch(project, branch)
-            _checkout(project, branch)
-
-
-class Fetch(WestCommand):
```

Comment:
I'm confused. In slack we said we could keep west fetch and west rebase.

```
carlesc [12:37 PM]
@mbolivar that really depends. Would fetch and rebase take an argument? `west fetch {yourfork,upstream}` `west rebase {yourfork/branch,upstream/master}`? without a clear definition of remote/revision like we had before I am not sure if I know the answer to your question

mbolivar [12:37 PM]
no changes to the current signature
`west fetch [-h] [--no-update] [PROJECT [PROJECT ...]]`

carlesc [12:38 PM]
ok, and and where would it fetch from? I think I need to know that to answer

mbolivar [12:38 PM]
`west rebase [-h] [PROJECT [PROJECT ...]]`

carlesc [12:38 PM]
in order to see if I find this intuitive
and what would I rebase onto?

mbolivar [12:38 PM]
in both cases the answer is 'the remote revision'
using the discussion we had earlier for how that's defined for project versus manifest repositories

carlesc [12:40 PM]
can you refresh my memory for the manifest repo remote revision? I am juggling many things apologies :slightly_smiling_face:

mbolivar [12:41 PM]
of course
same here :slightly_smiling_face:
1. if the manifest repo is detached, the remote revision is HEAD
2. if the manifest repo branch is tracking a remote branch, that's the remote revision
3. if the manifest repo branch is not tracking a remote branch, the remote revision is HEAD
basically we just use git to try to find a remote revision and default to HEAD otherwise

carlesc [12:45 PM]
ok understood. Then my answer to your question is yes, we should do so, with the following caveat: either as part of those commands or, I believe even better, through a `projects` meta command or similar, we should be able to do the same but leaving the manifest repo untouched
included in my answer is also the fact that I think that is also on your mind: we should drop the manifest url/rev passed to west init

mbolivar [12:46 PM]
yes, i think that's implied
i don't like a projects meta command tbh
parsing is already complicated enough
we are stretching argparse to its limits as it is; if we go any farther we are getting into "fork it" territory

carlesc [12:47 PM]
ok, then `--projects` ?
or maybe `-p/--projects-only`?
is that reasonable?

mbolivar [12:48 PM]
sure, modulo name bikeshedding, i think an option that says "just operate on the non-manifest projects" makes sense (edited) 

carlesc [12:52 PM]
ok. Just want to point something out: unless you use  `git branch -u` to change your upstream, the master upstream will always point to wherever  you cloned from, so you will always fetch from there.  This means that depending on whether you originally cloned from upstream or your own github fork of it you will fetch from one or the other. Both are quite common.
(we can document that)

mbolivar [1:14 PM]
this is for users converting to west, right?
users who bootstrap with west won't be cloning from their personal forks

carlesc [1:15 PM]
For anyone using west init
Why not? If they fork and then clone?

mbolivar [1:15 PM]
you're saying if they west init my-fork-url

carlesc [1:16 PM]
Exactly

mbolivar [1:16 PM]
i understand now. sure, but then hopefully it's clear from context

```

Reply:
sorry, maybe I was too quick in removing.
I was just looking at the list @carlescufi  made earlier.

Will put those two commands back in, but restrict them from working on manifest repo

Reply:
That above discussion is about how they should apply to the manifest repo too

Reply:
If we don't have time to support them now I suppose we could remove them and open an issue to add them back with the above discussion as the implementation strategy. Your call

Reply:
I think it should be possible, I just rebase, and don't delete the existing command.
The adjustment to manifest repo to fetch / rebase default remote (usually origin) should not be complicated.

Reply:
@mbolivar 
That conversation took place before the actual decision to remove those during a call. In fact the list we settled on was:

```
   Init(), - > same
    List(), -> Operate only on projects, except maybe if “-a”
    Clone(), -> Delete
    Fetch(), -> Delete
    Pull(), -> Delete
    Rebase(), -> Delete
    Branch(), -> Delete
    Checkout(), -> Delete
    Diff(), -> Operate only on projects, except maybe if “-a”
    Status(), -> Operate only on projects, except maybe if “-a”
    Update(), -> New command, old update becomes “west update --self-only”
    ForAll(), -> Operate only on projects, except maybe if “-a”
```

I do not think we have the time now to add those commands, especially since it is very likely that additional questions are raised during the addition.
My strong preference is to remove those commands for now, re-introduce later once west is in the tree.

Reply:
Sorry, as discussed on slack, I must have forgotten this. I'm ok with dropping these.

Reply:
Yep, that was the list I was using.
For a second I thought the slack discussion came after this list.

Although:
```
“west update --self-only”
```
can be executed as
```
west selfupdate
```
as it will be a cleaner cut to completely remote the command, instead of changing signature of an existing command, when `west` goes fully `pip`